### PR TITLE
Fail to load singe rulegroup cause rulers to halt the sync of all rulegroups.

### PR DESCRIPTION
**Pull request by alanprot**
**What this PR does**:
Implementing what is suggested in https://github.com/cortexproject/cortex/issues/4180#issuecomment-840573262

**Which issue(s) this PR fixes**:
Fixes: https://github.com/cortexproject/cortex/issues/4180

**Checklist**
- [X] Tests updated
- [ ] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ruler/rulestore/bucketclient/bucket_client.go [hunk 1]
```diff
@@ -61,6 +62,11 @@ func (b *BucketRuleStore) getRuleGroup(ctx context.Context, userID, namespace, g
 		return nil, rulestore.ErrGroupNotFound
 	}
 
+	if userBucket.IsAccessDeniedErr(err) {
+		level.Debug(b.logger).Log("msg", "permission denied when loading group", "user", userID, "key", objectKey)
```
Comment (yeya24) [inline pkg/ruler/rulestore/bucketclient/bucket_client.go:66]: Do we want to keep it debug level or use warn or something else?

File: pkg/ruler/rulestore/bucketclient/bucket_client.go [hunk 2]
```diff
@@ -177,17 +186,29 @@ func (b *BucketRuleStore) LoadRuleGroups(ctx context.Context, groupsToLoad map[s
 			for gr := range ch {
 				user, namespace, group := gr.GetUser(), gr.GetNamespace(), gr.GetName()
 				if user == "" || namespace == "" || group == "" {
-					return fmt.Errorf("invalid rule group: user=%q, namespace=%q, group=%q", user, namespace, group)
+					m.Lock()
+					e = fmt.Errorf("invalid rule group: user=%q, namespace=%q, group=%q", user, namespace, group)
```
Comment (yeya24) [inline pkg/ruler/rulestore/bucketclient/bucket_client.go:190]: Is it better to use multi err rather than just wrapping errors?

Comment (alanprot) [reply pkg/ruler/rulestore/bucketclient/bucket_client.go:190]: +1.. will do that

File: pkg/ruler/rulestore/bucketclient/bucket_client.go (hunk 1 above)
Comment (alanprot) [reply pkg/ruler/rulestore/bucketclient/bucket_client.go:66]: i think its fine as we will bubble up and log anyway on the ruler. WDYT?

Comment (yeya24) [reply pkg/ruler/rulestore/bucketclient/bucket_client.go:67]: I am good with debug, just curious

**Review by yeya24 — APPROVED** [review]

**Comment by qinxx108** [comment]
LGTM

**Pull request merged**